classes/class-log.php

```diff
@@ -87,9 +73,16 @@ public function log( $connector, $message, $args, $object_id, $context, $action,
 			return false;
 		}
 
+		$ip_address = $this->plugin->get_client_ip_address();
+
+		// Fallback to unsafe IP extracted from the request HTTP headers.
+		if ( empty( $ip_address ) ) {
```

Comment:
The `empty( $ip_address )` will be true for both the case where no IP address was provided as well for when an invalid IP address is provided. Is that the intended behavior?

Reply:
It is a bit tricky here -- we're attempting to preserve the `HTTP_X_FORWARDED_FOR` usage from before but now it will only happen if `REMOTE_ADDR` is not configured which might not be the case making this an equally breaking change.

I'm thinking we should probably do a breaking change release and make it consistent immediately to avoid confusion.